extensions/notion/src/create-database-page.tsx

```diff
@@ -0,0 +1,199 @@
+  // Get preference values
+  const notion_token = String(preferences.notion_token.value)
+  const notion_workspace_slug = String(preferences.notion_workspace_slug.value)
+  if (notion_token.length !== 50) {
+    showToast(ToastStyle.Failure, 'Invalid token detected')
+    throw new Error('Invalid token length detected')
+  }
+
+  // On form submit function
+  const { pop } = useNavigation();
+  async function handleSubmit(values: FormValues) {
+    if (!validateForm(values)) {
+      return;
+    }
+
+    setIsLoading(true)
+    const page = await createDatabasePage(values)
+    setIsLoading(false)
+    if(!page){
+      showToast(ToastStyle.Failure, 'Couldn\'t create database page');
+    }else{
+      showToast(ToastStyle.Success, 'Page created!');
+      pop();
+    }
+  }
+
+  // Setup useState objects
+  const [databases, setDatabases] = useState<Database[]>()
+  const [databaseProperties, setDatabaseProperties] = useState<DatabasePropertie[]>()  
+  const [databaseId, setDatabaseId] = useState<string>()
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  
+  // Fetch databases
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedDatabases = await fetchDatabases()
+     
+      setDatabases(fetchedDatabases)
+      setIsLoading(false)
+     
+    }
+    fetchData()
+  }, [])
+
+  // Fetch selected database property
+  useEffect(() => {
+    const fetchData = async () => {
+      if(databaseId){
+        setIsLoading(true)
+        const fetchedDatabaseProperties = await fetchDatabaseProperties(databaseId)     
+        setDatabaseProperties(fetchedDatabaseProperties)
+        setIsLoading(false)
+      }      
+    }
+    fetchData()
+  }, [databaseId])
+
+
+  if(databases && !databases[0]){
+    return NoSharedContent()
+  }
+  return (
+    <Form 
+      isLoading={isLoading} 
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <SubmitFormAction 
+              title='Create Page'
+              icon={Icon.Plus}
+              onSubmit={handleSubmit}
+               />
+            </ActionPanel.Section>
+        </ActionPanel>
+      }>
+      <Form.Dropdown 
+        key='database'
+        id='database'
+        title={'Database'}
+        onChange={setDatabaseId}>
+          {databases?.map((d) => {
+            return (
+              <Form.Dropdown.Item
+                key={d.id} 
+                value={d.id} 
+                title={(d.title ? d.title : 'Untitled') }
+                icon={((d.icon_emoji) ? d.icon_emoji : ( d.icon_file ?  d.icon_file :  ( d.icon_external ?  d.icon_external : Icon.TextDocument))) } />
+            )
+          })}
+      </Form.Dropdown>
+      <Form.Separator />
+      {databaseProperties?.map((dp) => {
+        const key = 'property::'+dp.type+'::'+dp.id;
+        const id = key;
+        const title = dp.name;
+        const placeholder = dp.type.replace(/_/g, ' ')
```

Comment:
Let's capitalize placeholder. It looks a bit weird with all these "title", "multi select", etc.